core/src/edu/cornell/gdiac/util/TiledJsonParser.java

```diff
@@ -57,4 +91,395 @@ public static void initPlatformFromJson(
     TextureRegion texture = new TextureRegion(directory.getEntry(key, Texture.class));
     obstacle.setTexture(texture);
   }
+
+  /**
+   * @param obstacle  obstacle to initialize
+   * @param directory asset directory for use of assets
+   * @param json      json to use for initialization
+   */
+  public static void initObjectFromJson(
+      SimpleObstacle obstacle, AssetDirectory directory, JsonValue json, Vector2 dScale,
+      int tHeight) {
```

Comment:
I like the idea of having a static method to initialize an object from JSON, and then having each model initialize its own specific properties. However, we should set this up in a different way. Instead of having each object have its own static method, we just modify the `initialize` method of each object to initialize its own custom JSON properties based on the JSON value. This way we don't have to make the setters of the properties for all objects public (having these as public kinda breaks some invariants). So I want you to refactor it in this way.

Actually, I think all `SimpleObstacle`s have an initialization method, so you should be able to call `obstacle.initialize` and then pass in the JsonValue to the object specific initialization method. Please lmk if you have questions about this implementation, I'd be happy to clarify on call if you're not sure how it should be structured!